DBZ-7461 Update doc for snapshot modes alignment (#5316)
debezium/debezium: documentation/modules/ROOT/pages/connectors/oracle.adoc

The connector performs an initial snapshot and then stops, without processing any subsequent changes.

`schema_only`:: Deprecated, see `no-data`

`no_data`:: The snapshot includes only the structure of captured tables.
Specify this value if you want the connector to capture data only for changes that occur after the snapshot.

`schema_only_recovery`:: Deprecated, see `recovery`.

`recovery`:: This is a recovery setting for a connector that has already been capturing changes.
When you restart the connector, this setting enables recovery of a corrupted or lost database schema history topic.
You might set it periodically to "clean up" a database schema history topic that has been growing unexpectedly.
Database schema history topics require infinite retention.
Note this mode is only safe to be used when it is guaranteed that no schema changes happened since the point in time the connector was shut down before and the point in time the snapshot is taken.

After the snapshot is complete, the connector continues to read change events from the database's redo logs except when `snapshot.mode` is configured as `initial_only`.

> `when_needed`:: The connector performs a snapshot upon startup whenever it deems it necessary. That is, when no offsets are available, or when a previously recorded offset specifies a log position that is not available in the server.

Reviewer: ```suggestion
`when_needed`:: After the connector starts, it performs a snapshot only if it detects one of the following circumstances:
 
* It cannot detect any topic offsets.
* A previously recorded offset specifies a log position that is not available on the server.
```

`none`:: Prevents the connector from acquiring any table locks during the snapshot.
  Use this setting only if no schema changes might occur during the creation of the snapshot.

ifdef::community[]
`custom`:: The connector performs a snapshot according to the implementation specified by the xref:oracle-property-snapshot-locking-mode-custom-name[`snapshot.locking.mode.custom.name`] property, which is a custom implementation of the `io.debezium.spi.snapshot.SnapshotLock` interface.
endif::community[]

ifdef::community[]
|[[oracle-property-snapshot-locking-mode-custom-name]]<<oracle-property-snapshot-locking-mode-custom-name, `+snapshot.locking.mode.custom.name+`>>
|No default
| When `snapshot.locking.mode` is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotLock' interface.

> For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].

Reviewer: Ditto.

endif::community[]

|[[oracle-property-snapshot-query-mode]]<<oracle-property-snapshot-query-mode, `+snapshot.query.mode+`>>
|`select_all`
|Specifies how the connector queries data while performing a snapshot. +
Set one of the following options:

> `select_all`:: The connector performs a `select all` query.

Reviewer 2: Same as above.

ifdef::community[]
`custom`:: The connector performs a snapshot query according to the implementation specified by the xref:oracle-property-snapshot-snapshot-query-mode-custom-name[`snapshot.query.mode.custom.name`] property, which defines a custom implementation of the `io.debezium.spi.snapshot.SnapshotQuery` interface. +
endif::community[]

This setting enables you to manage snapshot content in a more flexible manner compared to using the xref:oracle-property-snapshot-select-statement-overrides[`snapshot.select.statement.overrides`] property.

ifdef::community[]
|[[oracle-property-snapshot-snapshot-query-mode-custom-name]]<<oracle-property-snapshot-query-mode-custom-name, `+snapshot.query.mode.custom.name+`>>
|No default

> | When xref:oracle-property-snapshot-query-mode[`snapshot.query.mode`] is set as `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotQuery' interface.

Reviewer: ```suggestion
| When xref:oracle-property-snapshot-query-mode[`snapshot.query.mode`] is set to `custom`, use this setting to specify the name of the custom implementation provided in the `name()` method that is defined by the 'io.debezium.spi.snapshot.SnapshotQuery' interface.
```

> For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].

Reviewer: Ditto.

|Specifies how the connector queries data while performing a snapshot. +
Set one of the following options:

> `select_all`:: `select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.

Reviewer 2: ```suggestion
`select_all`:: The connector performs a `select all` query by default, optionally adjusting the columns selected based on the column include and exclude list configurations.
```
Same?
